src/__tests__/contextAndMirage.test.tsx

```diff
@@ -0,0 +1,201 @@
+        setQuantity((prevQuantity) => prevQuantity + 1);
+    }
+
+    async function handleDecrement () {
+        setQuantity((prevQuantity) => prevQuantity > 0 ? prevQuantity - 1 : 0)
+    }
+
+    async function handleAddToCart () {
+        const product = fetchedData[0];
+        if (quantity <= 0) return;
+        
+        addToCart({...product, quantity: quantity});
+    }
+
+    async function handleDeleteFromCart () {
+        const product = fetchedData[0];
+        deleteFromCart(product.id);
+    }
+
+    if (isLoading) return (<p>TestComponent is loading...</p>)
+
+    return (
+        <div>
+            <h1>Loading finished</h1>
+            <div>
+                <p>{fetchedData[0].company}</p>
+                <button onClick={handleIncrement}>+</button>
+                <button onClick={handleDecrement}>-</button>
+                <button onClick={handleAddToCart}>Add to cart</button>
+                <button onClick={handleDeleteFromCart}>Delete from cart</button>
+                <Image 
+                src={fetchedData[0].thumbnailImage} 
+                alt="thumbnail image"
+                />
+                <p id="quantity-holder">{quantity}</p>
+            </div>
+            {
+                cart.length > 0 && (
+                <div className='cart'>
+                    <h2>Cart</h2>
+                    <p id='cart-id'>Product ID is {cart[0].id}</p>
+                    <p id='cart-quantity' role="show-quantity">Quantity is {cart[0].quantity}</p>
+                </div>
+                )
+            }
+        </div>
+    )
+}
+
+describe('Test', () => {
+    let mirageServer: any;
+    beforeEach(() => {
+        mirageServer = makeServer()
+        render(
+            <CartProvider>
+                <TestComponent />
+            </CartProvider>
+        )
+    })
+
+    afterEach(() => {
+        mirageServer.shutdown()
+    })
+
+
+    it('fetches from API', async () => {
+        await waitFor(() => {
+            expect(screen.getByText('Loading finished')).toBeInTheDocument()
+        })
+
+        expect(screen.getByText("Sneaker Company")).toBeInTheDocument()
+    })
+
+    it('images are correctly imported in mirage', async () => {
+        // Wait for the images to be loaded
+        await waitFor(() => {
+            expect(screen.getByAltText('thumbnail image')).toBeInTheDocument()
+            const imgElement = screen.getByAltText('thumbnail image') as HTMLImageElement;
+            return imgElement && imgElement.complete;
+        });
+
+        // Check if the images are present in the document
+        const image = screen.getByAltText('thumbnail image') as HTMLImageElement;
+        expect(image.srcset).toContain('/_next/image?url')
+    })
+
+    it('quantity is incremented/decremented', async () => {
+        await waitFor(() => {
+            expect(screen.getByText('Loading finished')).toBeInTheDocument()
+        })
+
+        const incrementButton = screen.getByText('+');
```

Comment:
Instead of getting the text, what do you think of using a testId instead?
eg: 
```
element -> <button onClick={handleIncrement} data-testid="increment-button">+</button>
const incrementButton = screen.getByTestId('increment-button');
```
 you can check this for reference https://testing-library.com/docs/queries/bytestid/
 What do you think?

Reply:
That is indeed a better way to do it, I've refactored the code to use data-testid.